### ISSUE TITLE
Accessing Interface var results in error

### ISSUE THREAD

**Issue by fknfilewalker** [shader-slang/slang#5800]
```csharp
interface ITest {
    property int value;
};

struct Test : ITest {
    int value;
};

RWStructuredBuffer<ITest> inputBuffer;
```

Accessing `inputBuffer[0].value` results in `(0): error 38101: 'this' expression can only be used in members of an aggregate type`

**Comment by csyonghe** [comment]
I created this test and it works fine with top of tree:

```
//TEST:COMPARE_COMPUTE(filecheck-buffer=CHECK):-output-using-type

interface ITest {
    property int value;
};

struct Test : ITest {
    int value;
};

//TEST_INPUT: type_conformance Test:ITest = 1;

//TEST_INPUT: set inputBuffer = ubuffer(data=[0 0 1 0 1 0 0 0], stride=4)
RWStructuredBuffer<ITest> inputBuffer;

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer;

[NumThreads(1, 1, 1)]
void computeMain()
{
    // CHECK: 1
    outputBuffer[0] = inputBuffer[0].value;
}
```

Output is `1`.

**Comment by fknfilewalker** [comment]
This seems very strange. Sometimes it works, but some random code makes it not work.

When I use your exact code and add
```csharp
public interface TestInterface {
    public int test();
};
public struct Impl : TestInterface {
    public int test() { return 2; }
};
public struct Impl2 : TestInterface {
    public int test() { return 3;}
};
```
to it, it stops working on my side.
`(0): error 38101: 'this' expression can only be used in members of an aggregate type`

Removing the property from ITest makes it work again. It seems like adding property creates issues with certain other code?

**Comment by fknfilewalker** [comment]
```csharp
//TEST:COMPARE_COMPUTE(filecheck-buffer=CHECK):-output-using-type

public interface ITest {
    property int value;
};

struct Test : ITest {
    int value;
};

//TEST_INPUT: type_conformance Test:ITest = 1;

//TEST_INPUT: set inputBuffer = ubuffer(data=[0 0 1 0 1 0 0 0], stride=4)
RWStructuredBuffer<ITest> inputBuffer;

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer;

[NumThreads(1, 1, 1)]
void computeMain()
{
    // CHECK: 1
    outputBuffer[0] = inputBuffer[0].value;
}
```
does no longer work

**Comment by csyonghe** [comment]
Thank you for the repro. I have fixed this in PR #5817.
